### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by joao-carneiro-twistag** [twilio/twilio-voice.js#265]
- [ X ] I have verified that the issue occurs with the latest twilio.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-voice.js/blob/master/CHANGELOG.md).
 - [ X ] I reviewed the [Common Issues](https://github.com/twilio/twilio-voice.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [ X ] I verified that the [Quickstart](https://github.com/TwilioDevEd/voice-javascript-sdk-quickstart-node) application works in my environment.
 - [ X ] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

version: 2.10.2
This issue occurs sporadically and is challenging to reproduce consistently. 
However, I've observed that the TJS is generated instead of a valid callSid.
This seems to be related to the function `generateTempCallSid` in your repository.

**Expected behavior:**

Twilio should consistently return a valid callSid whenever a call is made.

**Actual behavior:**

In approximately 1 out of every 1000 calls, Twilio does not return a valid callSid. Instead, it provides a TJS.
I have extensive logs and reports available that can provide further insights into this issue upon request.

**Software versions:**

 - [ X ] Browser(s): Chrome, Firefox, Edge
 - [ ] Operating System: Windows
 - [ ] twilio.js: 2.10.2
 - [ ] Third-party libraries (e.g., Angular, React, etc.):

**Comment by joao-carneiro-twistag** [comment]
Also found this in your sdk.. 
![image](https://github.com/twilio/twilio-voice.js/assets/132991460/05717c0a-c15f-4a3b-b523-80997a1ba12c)

**Comment by charliesantos** [comment]
A a tempCallSid is used for outbound calls while the call is still trying to connect to Twilio. As soon as the call is established, usually after the ringing event, the real callsid will be returned by Twilio backend and it will be used. What you're seeing is normal.

**Comment by joao-carneiro-twistag** [comment]
I appreciate your answer. Yes, that makes sense, but,
I've encountered an issue on the client side of my application where, in some cases, long-duration calls lasting over 100 seconds do not result in receiving an original CallSid. This seems to occur particularly when calls are not answered.
So i end up with a lot of "lost" calls

**Comment by charliesantos** [comment]
Thanks @joao-carneiro-twistag . I took a look at the example tempCallSid above, TJSd65c394c-f3f8-4032-8a8d-a625dfb75736, it generated a real callSid CA76eaadfa0e73505d9e70b690e3049ad2. Where are you trying to read the real callsid from?

**Comment by joao-carneiro-twistag** [comment]
Thanks for your help @charliesantos ! I'm always reading it from parameters.CallSid

**Comment by charliesantos** [comment]
In that log you provided, the call is still trying to connect, so a real callsid is not available if you try to read it. For SDK outgoing calls, try reading it after the `call.on('ringing')` event or `call.on('accept')`. For SDK incoming calls, the real callSid is always available.

**Comment by joao-carneiro-twistag** [comment]
yes its an outgoing call and i'm reading it only after the event 'ringing', i'm afraid it could be due to internet issues, the client might lose the connection between the proccess

**Comment by charliesantos** [comment]
We can take a look more if you can reproduce the issue and provide the whole debug logs.

**Comment by kpchoy** [comment]
Closing this out due to inactivity. If you're still running into this, please open a new issue with the latest details and we'll take another look.
